x-element.d.ts

```diff
@@ -1,69 +1,239 @@
-export class XElement extends HTMLElement {
```

Comment:
Here’s a good example — this should have said `export default class XElement extends HTMLElement {` (see the `default`). Easy to miss since we often don’t depend on this module. Generating it from well-considered JSDoc “types” feels like a better way to ensure this stays accurate.